compiler/src/test/scala/dx/compiler/CompilerTest.scala

```diff
@@ -711,6 +721,149 @@ class CompilerTest extends AnyFlatSpec with Matchers with BeforeAndAfterAll {
+            )
+        )
+    )
+
+    val dxApplet2 = dxApi.applet(appId2)
+    val desc2 = dxApplet2.describe(
+        Set(
+            Field.Access
+        )
+    )
+
+    desc2.access shouldBe Some(
+        JsObject(
+            Map(
+                "allProjects" -> JsString("VIEW"),
+                "network" -> JsArray(Vector(JsString("*")))
+            )
+        )
+    )
+
+  }
+
+  it should "be able to build access for tasks in a workflow with dockerRegistry" in {
+    val path = pathFromBasename("non_spec", "dependency_report_wf2.wdl")
+    val extraPath = pathFromBasename("non_spec", "dependency_report_extras.json")
+    val args = path.toString :: "--extras" :: extraPath.toString :: cFlags
+    //:: "--verbose"
+    val wfId = Main.compile(args.toVector) match {
+      case SuccessfulCompileNativeNoTree(_, Vector(x)) => x
+      case other =>
+        throw new Exception(s"unexpected result ${other}")
+    }
+
+    val wf = dxApi.workflow(wfId)
+    val wfDesc = wf.describe(Set(Field.Stages))
+    val wfStages = wfDesc.stages.get
+
+    val fragAppletId =
+      wfStages.collectFirst({ case s if s.name == "frag dependency_report_t1" => s.executable }).get
+    val dxFragApplet = dxApi.applet(fragAppletId)
+    val fragDesc = dxFragApplet.describe(
+        Set(
+            Field.Access,
+            Field.Details
+        )
+    )
+    fragDesc.access shouldBe Some(
+        JsObject(
+            Map("allProjects" -> JsString("VIEW"), "network" -> JsArray(Vector.empty))
+        )
+    )
+    val taskAppletId = fragDesc.details match {
+      case Some(d) => {
+        val execLinkInfo = d.asJsObject.fields.getOrElse(
+            Constants.ExecLinkInfo,
+            throw new Exception(s"Missing ExecLinkInfo")
+        )
+        execLinkInfo match {
+          case JsObject(e) => {
+            val task = e.getOrElse(
+                "dependency_report_t1",
+                throw new Exception(s"Missing task dependency_report_t1")
+            )
+            task match {
+              case JsObject(t) => {
+                val taskId = t.getOrElse(
+                    "id",
+                    throw new Exception(s"Missing dependency_report_t1 applet id")
+                )
+                JsUtils.getString(taskId)
+              }
+              case _ => throw new Exception("Expected dependency_report_t1 applet info")
+            }
+          }
+          case _ => throw new Exception("Expected ExecLinkInfo as a map")
+        }
+      }
+      case _ => throw new Exception(s"Missing details")
+    }
+
+    val dxTaskApplet = dxApi.applet(taskAppletId)
+    val taskDesc = dxTaskApplet.describe(
+        Set(
+            Field.Access
+        )
+    )
+    taskDesc.access shouldBe Some(
+        JsObject(
+            Map(
+                "allProjects" -> JsString("VIEW"),
+                "network" -> JsArray(Vector(JsString("*")))
+            )
+        )
+    )
+
+    val dxNativeApp = dxApi.resolveApp(
```

Comment:
In this workflow we have 3 kinds of executables:

- frag applet: `network = []`
- task applet called by frag applet: `network = ["*"]` since it requires external docker
- native app: no change is applied to the native app